Gem/Code/Source/Automation/ScriptManager.cpp

```diff
@@ -1544,27 +1565,39 @@ namespace AtomSampleViewer
             // Note this will pause the script until the capture is complete
             if (PrepareForScreenCapture(imageName))
             {
-                AZStd::string screenshotFilePath;
+                AZ::Render::FrameCapturePathOutcome pathOutcome;
+
                 AZ::Render::FrameCaptureTestRequestBus::BroadcastResult(
-                    screenshotFilePath,
+                    pathOutcome,
                     &AZ::Render::FrameCaptureTestRequestBus::Events::BuildScreenshotFilePath,
                     imageName, true);
 
-                AZ_Assert(s_instance->m_frameCaptureId == AZ::Render::InvalidFrameCaptureId, "Attempting to start a capture while one is in progress");
-                AZ::Render::FrameCaptureId frameCaptureId = AZ::Render::InvalidFrameCaptureId;
-                AZ::Render::FrameCaptureRequestBus::BroadcastResult(frameCaptureId, &AZ::Render::FrameCaptureRequestBus::Events::CaptureScreenshot, screenshotFilePath);
-                if (frameCaptureId != AZ::Render::InvalidFrameCaptureId)
+                if (!pathOutcome.IsSuccess())
                 {
-                    s_instance->AZ::Render::FrameCaptureNotificationBus::Handler::BusConnect(frameCaptureId);
-                    s_instance->m_frameCaptureId = frameCaptureId;
+                    ReportScriptError(pathOutcome.GetError().m_errorMessage);
+                    s_instance->m_isCapturePending = false;
+                    s_instance->m_frameCaptureId = AZ::Render::InvalidFrameCaptureId;
+                    s_instance->ResumeScript();
+                    return;
                 }
-                else
+
+                AZStd::string screenshotFilePath = pathOutcome.GetValue();
+
+                AZ_Assert(s_instance->m_frameCaptureId == AZ::Render::InvalidFrameCaptureId, "Attempting to start a capture while one is in progress");
+                
+                AZ::Render::FrameCaptureOutcome capOutcome;
+                AZ::Render::FrameCaptureRequestBus::BroadcastResult(capOutcome, &AZ::Render::FrameCaptureRequestBus::Events::CaptureScreenshot, screenshotFilePath);
+                if (!capOutcome.IsSuccess())
                 {
-                    AZ_Error("Automation", false, "Failed to initiate frame capture for '%s'", screenshotFilePath.c_str());
+                    ReportScriptError(AZStd::string::format("Failed to initiate frame capture for '%s'", screenshotFilePath.c_str()));
                     s_instance->m_isCapturePending = false;
                     s_instance->m_frameCaptureId = AZ::Render::InvalidFrameCaptureId;
                     s_instance->ResumeScript();
+                    return;
                 }
+
+                s_instance->m_frameCaptureId = capOutcome.GetValue();
+                s_instance->AZ::Render::FrameCaptureNotificationBus::Handler::BusConnect(s_instance->m_frameCaptureId);
```

Comment:
This code is identical to the code above. Should be able to put in a function.

Reply:
Thanks for pointing this out. My next task is to merge ScriptManager into ScriptAutomation Gem, so that this part of the code will be reorganized. I'd skip refactoring in this PR and do a deeper one in the next PR.